components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/dao/AuthorizationCodeDAOImplTest.java

```diff
@@ -0,0 +1,446 @@
+import org.wso2.carbon.identity.oauth.config.OAuthServerConfiguration;
+import org.wso2.carbon.identity.oauth.dao.SQLQueries;
+import org.wso2.carbon.identity.oauth2.dao.util.DAOUtils;
+import org.wso2.carbon.identity.oauth2.internal.OAuth2ServiceComponentHolder;
+import org.wso2.carbon.identity.oauth2.model.AuthzCodeDO;
+import org.wso2.carbon.identity.oauth2.util.OAuth2TokenUtil;
+import org.wso2.carbon.identity.oauth2.util.OAuth2Util;
+import org.wso2.carbon.identity.testutil.powermock.PowerMockIdentityBaseTest;
+import org.wso2.carbon.user.core.UserCoreConstants;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.Timestamp;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyLong;
+import static org.mockito.Matchers.anyObject;
+import static org.mockito.Matchers.anyString;
+import static org.powermock.api.mockito.PowerMockito.doNothing;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+import static org.powermock.api.mockito.PowerMockito.when;
+import static org.wso2.carbon.identity.oauth.common.OAuthConstants.OAuthVersions.VERSION_2;
+import static org.wso2.carbon.identity.oauth2.util.OAuth2Util.getTenantId;
+
+/**
+ * Unit tests for AuthorizationCodeDAOImpl.
+ */
+@WithCarbonHome
+@PrepareForTest({IdentityDatabaseUtil.class, OAuth2Util.class, OAuth2TokenUtil.class, IdentityUtil.class,
+        OAuthServerConfiguration.class})
+public class AuthorizationCodeDAOImplTest extends PowerMockIdentityBaseTest {
+    public static Map<String, BasicDataSource> dataSourceMap = new HashMap<>();
+
+    @Mock
+    private ServiceProvider mockedServiceProvider;
+
+    @Mock
+    private AuthenticatedUser mockedAuthenticatedUser;
+
+    @Mock
+    private ApplicationManagementService applicationManagementService;
+
+    private Connection connection;
+
+    private AuthorizationCodeDAOImpl authorizationCodeDAO;
+    private String[] scopes;
+    AuthenticatedUser authenticatedUser = new AuthenticatedUser();
+    private static final int DEFAULT_TENANT_ID = 1234;
+    private static final String APP_NAME = "myApp";
+    private static final String USER_NAME = "user1";
+    private static final String CALLBACK = "http://localhost:8080/redirect";
+    private static final String DB_NAME = "testAuthzCODEDB";
+
+    @BeforeClass
+    public void initTest() throws Exception {
+
+        //Initializing the database.
+        DAOUtils.initializeDataSource(DB_NAME, DAOUtils.getFilePath("identity.sql"));
+        authorizationCodeDAO = new AuthorizationCodeDAOImpl();
+        scopes = new String[]{"sms", "openid", "email"};
+        authenticatedUser.setTenantDomain("super.wso2");
+        authenticatedUser.setUserName("randomUser");
+        authenticatedUser.setUserStoreDomain("PRIMARY");
```

Comment:
since the userstore domain is used in multiple places, let's define `"PRIMARY"` as a constant and use.

Reply:
Addressed with bd9a919ce82d35f572e275a65b173e7fdf9e50dc